invenio_search/ext.py

```diff
@@ -111,34 +112,38 @@ def _walk_dir(aliases, *parts):
         # Start the recursion here:
         _walk_dir(self.aliases, alias)
 
-    def register_templates(self, directory, package_name):
+    def register_templates(self, directory):
         """Register templates from the provided directory.
 
         :param directory: The templates directory.
         :param package_name: The package name.
         """
         result = {}
+        module_name, parts = directory.split('.')[0], directory.split('.')[1:]
+        parts = tuple(parts)
 
-        def _walk_dir(*parts):
+        def _walk_dir(parts):
             resource_name = os.path.join(*parts)
 
-            for filename in resource_listdir(package_name, resource_name):
+            for filename in resource_listdir(module_name, resource_name):
+                if ES_VERSION[0] == 2 and filename.startswith('v5'):
```

Comment:
This needs to be documented. If I understand correctly the function lists mappings for all versions and the wrong versions get filtered out.